http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/https/HTTPSProtocolsTest.java

```diff
@@ -0,0 +1,177 @@
+        return new Object[][] { { "TLSv1.1", "TLSv1.1", false, 9009 }, { "TLSv1.2", "TLSv1.1", true, 9008 },
+                { "TLSv1.1", "TLSv1.2", true, 9007 } };
+    }
+
+    @Test(dataProvider = "protocols")
+    /**
+     * Set up the client and the server
+     * @param clientProtocol SSL enabled protocol of client
+     * @param serverProtocol SSL enabled protocol of server
+     * @param hasException expecting an exception true/false
+     * @param serverPort port
+     */
+    public void setup(String clientProtocol, String serverProtocol, boolean hasException, int serverPort)
+            throws InterruptedException {
+
+        Parameter clientprotocols = new Parameter("sslEnabledProtocols", clientProtocol);
+        List<Parameter> clientParams = new ArrayList<>();
+        clientParams.add(clientprotocols);
+
+        Parameter serverProtocols = new Parameter("sslEnabledProtocols", serverProtocol);
+        List<Parameter> severParams = new ArrayList<>();
+        severParams.add(serverProtocols);
+
+        TransportsConfiguration transportsConfiguration = TestUtil
+                .getConfiguration("/simple-test-config" + File.separator + "netty-transports.yml");
+        Set<SenderConfiguration> senderConfig = transportsConfiguration.getSenderConfigurations();
+        senderConfig.forEach(config -> {
+            config.setKeyStoreFile(keyStoreFile);
+            config.setKeyStorePassword(password);
+            config.setParameters(clientParams);
+        });
+
+        HttpWsConnectorFactory factory = new HttpWsConnectorFactoryImpl();
+        ListenerConfiguration listenerConfiguration = ListenerConfiguration.getDefault();
+        listenerConfiguration.setPort(serverPort);
+        listenerConfiguration.setVerifyClient(verifyClient);
+        listenerConfiguration.setTrustStoreFile(trustStoreFile);
+        listenerConfiguration.setKeyStoreFile(keyStoreFile);
+        listenerConfiguration.setTrustStorePass(password);
+        listenerConfiguration.setKeyStorePass(password);
+        listenerConfiguration.setCertPass(password);
+        listenerConfiguration.setScheme(scheme);
+        listenerConfiguration.setParameters(severParams);
+        serverConnector = factory
+                .createServerConnector(ServerBootstrapConfiguration.getInstance(), listenerConfiguration);
+        ServerConnectorFuture future = serverConnector.start();
+        future.setHttpConnectorListener(new EchoMessageListener());
+        future.sync();
+        HttpWsConnectorFactory connectorFactory = new HttpWsConnectorFactoryImpl();
+        httpClientConnector = connectorFactory
+                .createHttpClientConnector(HTTPConnectorUtil.getTransportProperties(transportsConfiguration),
+                        HTTPConnectorUtil.getSenderConfiguration(transportsConfiguration, Constants.HTTPS_SCHEME));
+        testHttpsProtocols(hasException, serverPort);
+    }
+
+    public void testHttpsProtocols(boolean hasException, int serverPort) {
```

Comment:
instead of saying https protocols, shall we ssl protocols ?

Reply:
Done